### PR TITLE
Add RFC to disable Chromium stability checks

### PR DESCRIPTION
**Pull request by past**
This RFC follows a proposal from @jgraham in a discussion on Matrix from a few days ago.

### CODE REVIEW

**Comment by past** [comment]
In the WPT RFCs and infrastructure meeting yesterday I promised to share more details about how the flakiness detection works in Chromium CI. Here is a publicly accessible recent [example](https://ci.chromium.org/ui/p/chromium/builders/try/linux-rel/1287906/overview), courtesy of @WeizhongX. 

If you look closely you see that in step 45 we run blink_wpt_tests, which fail. Then in step 50 the system is trying to query the results database for any pre-existing flakiness. And it was able to decide that the failure in blink_web_tests (step 42) is due to a known flaky test, since there is a record of prior failed runs with that test (check the [stdout](https://logs.chromium.org/logs/chromium/buildbucket/cr-buildbucket/8789728943222392113/+/u/query_LUCI_Analysis_for_flaky_tests/stdout) link). It doesn't see any pre-existing flakiness in the failed blink_wpt_tests though. Then in step 57 it retries (perhaps unnecessarily) the shards with the patch a second time. In step 71, it retries without the patch applied. At this point, any previously failed tests will be run 10 times each. Any single failure here will cause the test to be treated as an existing failure, and the patch will be exonerated. It doesn't detect any failures, so the patch is clearly at fault. Finally in step 78 we can see that the patch was exonerated for the blink_web_tests failure since it was pre-existing, but in step 80 it declares the patch responsible for the blink_wpt_tests failures since reruns without the patch were successful.

In case it is useful, the recipe code for trybots can be found [here](https://source.chromium.org/chromium/chromium/tools/build/+/main:recipes/recipes/chromium_trybot.py) and the recipe documentation is [here](https://chromium.googlesource.com/infra/luci/recipes-py/+/HEAD/README.md) as well. There are also a few different internal dashboards and alerts that automate the investigation process and provide aggregate data to browser engineers. 

Of course all this is not easy to upstream to WPT infra, but hopefully it provides some pointers on what a future comprehensive solution could look like. For now, disabling the stability checks or making them non-blocking is the most efficient path.

File: rfcs/disable_chrome_stability.md
Comment (jgraham) [inline rfcs/disable_chrome_stability.md:15]: I don't think we should make claims about vendor CIs having better tooling in general. The reason we don't run them for PRs introduced by those browsers isn't that we expect they have _better_ tooling, but that we assume that a change which was considered stable enough to land in a vendor repo is also stable enough to land upstream, or at least that if it does cause flake in the vendor repo it will be fixed. In addition it's unusually likely that patches from vendor repos _add_ flaky tests at the same time as fixing the underlying issue. That's fine in the vendor CI where the build running the test will have the code changes, but since the builds we are running won't get those changes until they reach the relevant experimental channel, the added tests are likely to be unstable in GitHub CI.

**Review by jgraham — COMMENTED** [review]
I don't object to this or anything, but I think we should be clear that it's specifically about Chromium engineers preferring a different tradeoff to the status quo based on the properties of their own test system, and that the arguments don't readily generalise outside of that context.

As mentioned we also need to specifically pick a solution and outline what's required to implement that solution (e.g. to make Chrome checks non-blocking we actually need to change the taskcluster configuration so they don't cause the sink task to fail).

**Comment by foolip** [comment]
I just had https://github.com/web-platform-tests/wpt/pull/39105 blocked by this. I can admin merge myself, but most contributors can't and would be stuck for a while. @past do you have time soon to finish this?

**Comment by whimboo** [comment]
> I just had [web-platform-tests/wpt#39105](https://github.com/web-platform-tests/wpt/pull/39105) blocked by this. I can admin merge myself, but most contributors can't and would be stuck for a while. @past do you have time soon to finish this?

I can second that. I don't have admin permissions and as such cannot merge myself. We always have to ask @jgraham to get such PRs merged which in recent weeks is probably around 90% of the PRs. We as well would appreciate if this PR can be finished.

File: rfcs/disable_chrome_stability.md
Comment (past) [reply rfcs/disable_chrome_stability.md:15]: I don't think I am making any strong claims here, I said "_usually_ have better tools". Your comment suggests that _something_ makes doing the work downstream "better", but doesn't clearly identify that is. I tried to relax it further to "usually have better tools or processes", I hope that is satisfactory.

File: rfcs/disable_chrome_stability.md
Comment (past) [reply rfcs/disable_chrome_stability.md:29]: Agreed and text updated accordingly.

File: rfcs/disable_chrome_stability.md
Comment (past) [reply rfcs/disable_chrome_stability.md:23]: I don't understand what wording changes would satisfy you here, I believe we are in agreement about there being a tradeoff, which I describe in the Risks section. And to stress the point of who is happy and making that tradeoff, I only speak about the Chromium engineers in the entire RFC. Happy to amend further if you can provide a specific text.

**Comment by past** [comment]
Apologies for the delay, I believe I have now addressed the review feedback, so this is ready for another review.

**Comment by whimboo** [comment]
@past so who has to review these changes from Google's side? Looks like this PR is stuck.

Interestingly I have seen no such failure for quite some time for our PRs, but maybe I have missed such failures for others. Is there a way to easily see the failure rate of the Chromium stability job?

**Comment by jgraham** [comment]
I don't quite know what you were hoping for, but I've rewritten the RFC to better explain the tradeoffs as I understand them, and provide a bit more detail on the actual implementation requirements.

**Comment by past** [comment]
Thanks James, I'm happy with your changes if you want to review and merge this.

**Comment by jgraham** [comment]
Since this was a fairly substantive change I think it requires at least the one week timeout before merging. But also I'd be interested in input from other members of @web-platform-tests/wpt-core-team

**Review by Ms2ger — APPROVED** [review]

**Pull request merged**
